### PR TITLE
Run cmake-format over cmake files

### PR DESCRIPTION
**Pull request by dthaler**
Issue #2797 called out a couple of gaps, of which one is that the cmake files (*.cmake, CMakeLists.txt) do not have a consistent style.   This PR addresses that specific gap by using the cmake-format tool that Amaury pointed to in that issue.

This PR has 2 commits which should be reviewed separately, for efficiency of review.  The first commit is the actual change, and is relatively small and easy to check each line of (especially since most of the code in the scripts is identical to the existing code-format scripts).   The second commit is the result of actually running the tool, and is large but dynamically generated so reviewers need not review everything, just spot check it.

Since there were previously a mix of styles in the cmake files, please look at the second commit to see if the style it's converting to looks ok, and if not then the options in the .cmake-format.json file in the first commit are what controls the style.

### CODE REVIEW

**Comment by dthaler** [comment]
bors try

File: scripts/install-windows-prereqs.ps1 [hunk 1]
```diff
@@ -359,6 +359,10 @@ function Install-Python3 {
     New-Directory -Path $installDir -RemoveExisting
     Move-Item -Path "$tempInstallDir\*" -Destination $installDir
     Add-ToSystemPath -Path $installDir
+
+    Start-ExecuteWithRetry -ScriptBlock {
+        pip install cmake_format
```
Comment (jhand2) [inline scripts/install-windows-prereqs.ps1:364]: Does this script already install python?

Comment (dthaler) [reply scripts/install-windows-prereqs.ps1:364]: Yes, that's what line 354 does

**Comment by dthaler** [comment]
Rebased.  I don't think there's any requirement to hold other PRs.  This creates a new gate.  If anything sneaks in after the rebase and before the merge, then it's easy to submit a follow-up PR to clean up any stragglers.   Based on 3 approvals, I think this can be merged now.

**Comment by dthaler** [comment]
bors retry

**Pull request merged**